istioctl/pkg/multicluster/remote_secret.go

```diff
@@ -238,6 +238,74 @@ func getServiceAccountSecret(client kube.ExtendedClient, opt RemoteSecretOptions
 		return nil, err
 	}
 
+	if !kube.IsAtLeastVersion(client, 24) {
+		return legacyGetServiceAccountSecret(serviceAccount, client, opt)
+	}
+	return getOrCreateServiceAccountSecret(serviceAccount, client, opt)
+
+}
+
+// In 1.24+ we can't assume the secrets will be referenced in the ServiceAccount or be created automatically.
+// See https://github.com/istio/istio/issues/38246
+func getOrCreateServiceAccountSecret(
+	serviceAccount *v1.ServiceAccount,
+	client kube.ExtendedClient,
+	opt RemoteSecretOptions,
+) (*v1.Secret, error) {
+	ctx := context.TODO()
+
+	// manually specified secret, make sure it references the ServiceAccount
+	if opt.SecretName != "" {
+		secret, err := client.CoreV1().Secrets(opt.Namespace).Get(ctx, opt.SecretName, metav1.GetOptions{})
+		if err != nil {
+			return nil, fmt.Errorf("could not get specified secret %s/%s: %v",
+				opt.Namespace, opt.SecretName, err)
+		}
+		if err := validateServiceAccountSecret(serviceAccount, secret); err != nil {
```

Comment:
Do we need something added to the analyzer for this as well?

Reply:
I don't think so. 

1. We may not even use static tokens in some meshes. 
2. We really only read from this secret during remote-secret creation. The remote-secret won't have explicit references to this, just extracted content.